modules/aws/vpc/sg-etcd.tf

```diff
@@ -6,44 +6,64 @@ resource "aws_security_group" "etcd" {
       "kubernetes.io/cluster/${var.cluster_name}", "owned",
       "tectonicClusterID", "${var.cluster_id}"
     ), var.extra_tags)}"
+}
+
+resource "aws_security_group_rule" "etcd_egress" {
+  type              = "egress"
+  security_group_id = "${aws_security_group.etcd.id}"
+
+  from_port   = 0
+  cidr_blocks = ["0.0.0.0/0"]
+  to_port     = 0
+  protocol    = "-1"
+}
+
+resource "aws_security_group_rule" "etcd_ingress_icmp" {
+  type              = "ingress"
+  security_group_id = "${aws_security_group.etcd.id}"
+
+  protocol    = "icmp"
+  cidr_blocks = ["0.0.0.0/0"]
+  from_port   = 0
+  to_port     = 0
+}
+
+resource "aws_security_group_rule" "etcd_ingress_ssh" {
+  type              = "ingress"
+  security_group_id = "${aws_security_group.etcd.id}"
+
+  protocol  = "tcp"
+  from_port = 22
+  to_port   = 22
+  self      = true
+}
+
+resource "aws_security_group_rule" "etcd_ingress_etcd" {
+  type              = "ingress"
+  security_group_id = "${aws_security_group.etcd.id}"
+
+  protocol  = "tcp"
+  from_port = 2379
+  to_port   = 2379
+  self      = true
+}
+
+resource "aws_security_group_rule" "etcd_ingress_peer" {
+  type              = "ingress"
+  security_group_id = "${aws_security_group.etcd.id}"
+
+  protocol  = "tcp"
+  from_port = 2380
+  to_port   = 2380
+  self      = true
+}
+
+resource "aws_security_group_rule" "etcd_ingress_kubelet_insecure_from_master" {
+  type                     = "ingress"
+  security_group_id        = "${aws_security_group.etcd.id}"
+  source_security_group_id = "${aws_security_group.master.id}"
 
-  egress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = "-1"
-    self        = true
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-
-  ingress {
-    protocol    = "icmp"
-    cidr_blocks = ["0.0.0.0/0"]
-    from_port   = 0
-    to_port     = 0
-  }
-
-  ingress {
-    protocol  = "tcp"
-    from_port = 22
-    to_port   = 22
-    self      = true
-
-    security_groups = ["${aws_security_group.master.id}"]
-  }
-
-  ingress {
-    protocol  = "tcp"
-    from_port = 2379
-    to_port   = 2379
-    self      = true
-
-    security_groups = ["${aws_security_group.master.id}"]
-  }
-
-  ingress {
-    protocol  = "tcp"
-    from_port = 2380
-    to_port   = 2380
-    self      = true
-  }
+  protocol  = "tcp"
+  from_port = 10250
```

Comment:
Are all of the 'from' correct? don't most things trying to connect to here use a random high port?

Reply:
> Are all of the 'from' correct? don't most things trying to connect to here use a random high port?

Yeah, that doesn't make sense to me either.  But it's what we have had in master since forever; see, for example, [here][1].  Still, I'll drop them and see if that helps.

[1]: https://github.com/openshift/installer/commit/b620c16f0a4b82ee65b2d248fd98512a9f7e430e#diff-952765b0711a2c98e18e383033131664R27

Reply:
> > Are all of the 'from' correct? don't most things trying to connect to here use a random high port?
[
> Yeah, that doesn't make sense to me either...

Ah, [`from_port`][1] and [`to_port`][2] are for a range of ports on a single host, not the ports for both hosts involved in the connection.  So having `from_port` makes sense.

[1]: https://www.terraform.io/docs/providers/aws/r/security_group_rule.html#from_port-1
[2]: https://www.terraform.io/docs/providers/aws/r/security_group_rule.html#to_port-1